src/fu-util.c

```diff
@@ -727,6 +732,62 @@ fu_util_report_history_for_uri (FuUtilPrivate *priv,
 				  SOUP_MEMORY_COPY, data, strlen (data));
 	status_code = soup_session_send_message (priv->soup_session, msg);
 	g_debug ("server returned: %s", msg->response_body->data);
+
+	/* parse JSON reply */
+	json_parser = json_parser_new ();
+	if (!json_parser_load_from_data (json_parser,
+					 msg->response_body->data,
+					 msg->response_body->length,
+					 error)) {
+		g_prefix_error (error, "Failed to parse JSON response: ");
```

Comment:
Not familiar enough with `json_parser_load_from_data` to know, but if it doesn't show what the response it was  trying to parse into the `GError`, probably should add it here.

Reply:
Fixed.